libraries/AP_BattMonitor/AP_BattMonitor_DJIBattery.h

```diff
@@ -0,0 +1,37 @@
+#pragma once
+
+#include "AP_BattMonitor.h"
+#include "AP_BattMonitor_Backend.h"
+
+class AP_BattMonitor_DJIBattery : public AP_BattMonitor_Backend
+{
+public:
+
+    /// Constructor
+    AP_BattMonitor_DJIBattery(AP_BattMonitor &mon, AP_BattMonitor::BattMonitor_State &mon_state, AP_BattMonitor_Params &params);
+
+    /// Read the battery voltage and percentage.
+    void read() override;
+
+    /// returns true if battery monitor provides consumed energy info
+    bool has_consumed_energy() const override { return true; }
```

Comment:
You don't actually have this yet, because you need to calculate it. Either remove it, or add the runtime calculation (you'd basically be mimicking the way the analog battery monitor finds this information).

Reply:
is there any other way or parameter to show battery percentage? please help me regarding this.